internal/backend/runtime/kubernetes/kubernetes.go

```diff
@@ -122,25 +118,57 @@ func (r *Runtime) Get(ctx context.Context, dest interface{}, setters ...runtime.
 			Name:      opts.Name,
 		}, object)
 	} else {
-		selector := labels.NewSelector()
+		return nil, fmt.Errorf("both resource type and resource name must be defined for Kubernetes Get")
+	}
 
-		if opts.LabelSelector != "" {
-			selector, err = labels.Parse(opts.LabelSelector)
-			if err != nil {
-				return err
-			}
+	if err != nil {
+		return nil, err
+	}
+
+	return object, nil
+}
+
+// List implements runtime.Runtime.
+func (r *Runtime) List(ctx context.Context, setters ...runtime.QueryOption) (interface{}, error) {
+	opts := runtime.NewQueryOptions(setters...)
+
+	// turn single item into a list
+	if opts.Resource != "" {
+		parts := strings.Split(opts.Resource, ".")
+		if !strings.HasSuffix(parts[0], "List") {
+			parts[0] += "List"
```

Comment:
Maybe there's a better way, but gosh... Kubernetes clients are driving me nuts. Need to retry looking for that with a fresh head.